[FEATURE] Add hook to modify fetched content

adrienjacob opened the pull request:
This hook allows to use external extensions, changing the content (e.g. links) of fetched content.

adrienjacob committed:
[FEATURE] Add hook to modify fetched content

This hook allows to use external extensions, changing the content (e.g. links) of fetched content.

einpraegsam commented:
A hooking possibility is a good idea but I would prefer a signal instead of an old hook - what do you think? Next thing is, I would place the signal not in getBodyContent() but directly in fetchurl().

adrienjacob committed:
Change from hook to signal

The signal is called like this
$signalSlotDispatcher =
\TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Extbase\SignalSlot\Dispatcher::class);
$signalSlotDispatcher->connect(
'In2code\\Fetchurl\\Domain\\Service\\FetchService',
'fetchUrl',
'Path\\To\\Extension',
'extensionFunction'
);

adrienjacob commented:
Alright then, it's probably more future proof indeed. I made the change in the last commit...
